src/test/java/com/google/sps/servlets/UserDataServletTest.java

```diff
@@ -0,0 +1,256 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+/** Tests the User Data Servlet */
+@RunWith(JUnit4.class)
+public final class UserDataServletTest {
+  private static final String USER_BIO_PROPERTY = "bio";
+  private static final String USER_EMAIL_PROPERTY = "email";
+  private static final String USER_FOUND_PROPERTY = "user-found";
+  private static final String USER_FRIENDS_LIST_PROPERTY = "friends-list";
+  private static final String USER_ID_PROPERTY = "id";
+  private static final String USER_NAME_PROPERTY = "name";
+
+  // Test User info
+  private static final String TEST_USER_NAME = "Tim";
+  private static final String TEST_USER_ID = "123";
+  private static final String TEST_USER_EMAIL = "tim@gmail.com";
+  private static final String TEST_USER_BIO = "Amazing!";
+  private static final String[] TEST_USER_FRIENDS_LIST = new String[]{"321"};
+  private static final String ALTERNATE_TEST_USER_NAME = "John";
+
+  private final Gson gson = new Gson();
+  // Uses a local datastore stored in memory for tests
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Mock private HttpServletRequest mockRequest;
+  @Mock private HttpServletResponse mockResponse;
+  private UserDataServlet servletUnderTest;
+  private StringWriter stringWriter;
+  private PrintWriter writer;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    servletUnderTest = new UserDataServlet();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  /**
+   * Tests the doGet method, where the datastore doesn't contain the test user id.
+   * Expected output: JSON response with "user-found" property as false.
+   */
+  @Test
+  public void testGetMethodWithInvalidId() throws Exception {
+    // Mock the getParameter call
+    when(mockRequest.getParameter(USER_ID_PROPERTY)).thenReturn(TEST_USER_ID);
+
+    // Store output given by mockResponse
+    stringWriter = new StringWriter();
+    writer = new PrintWriter(stringWriter);
+    when(mockResponse.getWriter()).thenReturn(writer);
+    when(mockResponse.getContentType()).thenReturn("application/json");
+
+    // Create a local datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    // Test the doGet method by calling the wrapper method
+    servletUnderTest.doGetWrapper(datastore, mockRequest, mockResponse);
+    writer.flush();
+
+    // Build the correct response output
+    Map<String, Object> correctResponse = new HashMap<>();
+    correctResponse.put(USER_FOUND_PROPERTY, false);
+
+    // Check that the user wasn't found in datastore
+    verify(mockRequest, atLeast(1)).getParameter(USER_ID_PROPERTY);
+    Assert.assertEquals(gson.toJson(correctResponse), stringWriter.toString().trim());
```

Comment:
Sounds good, Truth looks like it provides better assertions. I'm using the `contains` method for `StringSubject` (returned by `assertThat`) because it looked to be the most suitable method, especially since the mockRequest returns a JSON string.